### DISCUSSION TITLE
I need to open H2SONG old files from 2015. Can anyone help?

### DISCUSSION THREAD

**Discussion by sombra-11** [hydrogen-music/hydrogen#2078] [Hydrogen on Windows]
Hi! I have H2SONG files that I was working on back in 2015, but they won't play on the latest version of Hydrogen (they will open, but there's no sound when i press play). I can't find an earlier version that works. Can anybody please help me with that? Thanks in advance

**Comment by theGreatWhiteShark** [comment]
Hey @sombra-11 ,

Sure. Can you upload the song file?

**Reply by sombra-11** [reply]
Thanks for replying! I tried to upload, but it says the file type is not supported here. But I have a LOT of H2SONG files from back then, not just one. That's why I want to find a way to play them.

This is the message I got when trying to upload: "We don’t support that file type. Try again with GIF, JPEG, JPG, MOV, MP4, PNG, SVG, WEBM, CPUPROFILE, CSV, DMP, DOCX, FODG, FODP, FODS, FODT, GZ, JSON, JSONC, LOG, MD, ODF, ODG, ODP, ODS, ODT, PATCH, PDF, PPTX, TGZ, TXT, XLS, XLSX or ZIP."

**Reply by trebmuh** [reply]
Just rename the file with .txt before uploading, and that would do.

**Reply by theGreatWhiteShark** [reply]
> But I have a LOT of H2SONG files from back then, not just one. That's why I want to find a way to play them.

There is a high probability this is just about a drumkit missing or a sample moved. Fixing one file might result in fixing all.

**Reply by sombra-11** [reply]
Thanks guys! I renamed it with .txt and tried to upload but I still get the same error message :(

I think you're right about the drumkit missing problem.

**Reply by theGreatWhiteShark** [reply]
> I renamed it with .txt and tried to upload but I still get the same error message :(

Could you put it into an archive? Both `.zip` and `.tar.gz` will do.

**Reply by sombra-11** [reply]
Thanks! I think that worked
[random beat.txt.zip](https://github.com/user-attachments/files/18119133/random.beat.txt.zip)

**Reply by theGreatWhiteShark** [reply]
> Thanks! I think that worked

Just tried it and it loads fine with latest version 1.2.4.

The song expects the kit `HipHop-2` to be present in `C:/Users/eduardo/.hydrogen/data/drumkits/HipHop-2`. The kit itself is still available via the online import. But since the song was created with a rather old version, Hydrogen can not fall back to the kit being located in a different location. The paths encoded in the songs have to match. So, you could either open them using an "eduardo" user or make a search-and-replace for the user name in all song using your favorite editor.

**Reply by sombra-11** [reply]
Thank you very much! Today I'm kinda busy, but I'll try that soon.
